### ISSUE TITLE
dropbox backend fails with "backslash not allowed" error since get_available_name method is not overrided

### ISSUE THREAD

**Issue by epetxepe** [jschneier/django-storages#137]
Hi all, 

  I'm using django version 1.9, python 2.7. 

  I've set up dropbox backend and configured the Oauth token. I'm using django-storages for a FileField of my model.

  I'm getting:

```
ErrorResponse: [400] {u'path': u"Invalid path '/biocloud\\b2ce86b5-dffb-4752-b7c3-c9c27d9d40fd\\18.jpg': character at index 9: backslash not allowed"} 
```

  I've tried to convert the paths at django model level, but it gets converted back when django starts the saving process.

  I'm using S3 backend as an alternative, so I've done some debugging on how django actually executes both.

  S3 backend for django-storages overrides get_available_name, and does the backslash => slash conversion.

  dropbox for django-storages does not override get_available_name and the default django method, appends some alphanumeric if already exists or truncates path if it is > max_length. So no conversion is performed.  And return force_text(name.replace('\', '/')) at the end of django's storage.py save() is only applied to return a name to be saved in the model, but not to be send to the dropbox storage backend logic for the overrided "_save ()" function.

 Right now I've overcome this issue by adding "name = name.replace ('\','/')" at the beginning of dropbox backend "_save ()" function but I don't think is the best option and I think that get_available_name should be overrided in the dropbox backend.

Regards,

Gorka

**Comment by jschneier** [comment]
Hi and thanks for the report! Would you be interested in creating a pull request to fix this issue?

**Comment by jschneier** [comment]
I think this is fixed by e52a127523fdd5be50bb670ccad566c5d527f3d1.
